gcloud-java-compute/src/main/java/com/google/gcloud/compute/Zone.java

```diff
@@ -0,0 +1,380 @@
+  }
+
+  /**
+   * Returns the zone's identity.
+   */
+  public ZoneId zoneId() {
+    return zoneId;
+  }
+
+  /**
+   * Returns the creation timestamp in RFC3339 text format.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc3339.txt">RFC3339</a>
+   */
+  public String creationTimestamp() {
+    return creationTimestamp;
+  }
+
+  /**
+   * Returns an optional textual description of the zone.
+   */
+  public String description() {
+    return description;
+  }
+
+  /**
+   * Returns an unique identifier for the zone; defined by the service.
+   */
+  public Long id() {
+    return id;
+  }
+
+  /**
+   * Returns a service-defined URL for the zone.
+   */
+  public String selfLink() {
+    return selfLink;
+  }
+
+  /**
+   * Returns the status of the zone.
+   */
+  public Status status() {
+    return status;
+  }
+
+  /**
+   * Returns the scheduled maintenance windows for this zone, if any. When the zone is in a
+   * maintenance window, all resources which reside in the zone will be unavailable.
+   *
+   * @see <a href="https://cloud.google.com/compute/docs/robustsystems#maintenance">Maintenance
+   *     Windows</a>
+   */
+  public List<MaintenanceWindow> maintenanceWindows() {
+    return maintenanceWindows;
+  }
+
+  /**
+   * Returns the identity of the region that hosts the zone.
+   */
+  public RegionId region() {
+    return region;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("zoneId", zoneId)
+        .add("id", id)
+        .add("creationTimestamp", creationTimestamp)
+        .add("description", description)
+        .add("selfLink", selfLink)
+        .add("status", status)
+        .add("maintenanceWindows", maintenanceWindows)
+        .add("region", region)
+        .toString();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(zoneId);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof Zone && Objects.equals(toPb(), ((Zone) obj).toPb());
+  }
+
+  com.google.api.services.compute.model.Zone toPb() {
+    com.google.api.services.compute.model.Zone zonePb =
+        new com.google.api.services.compute.model.Zone();
+    zonePb.setId(BigInteger.valueOf(id));
+    zonePb.setCreationTimestamp(creationTimestamp);
+    zonePb.setName(zoneId.zone());
+    zonePb.setDescription(description);
+    zonePb.setSelfLink(selfLink);
+    zonePb.setStatus(status.name());
+    if (maintenanceWindows != null) {
+      zonePb.setMaintenanceWindows(
+          Lists.transform(maintenanceWindows, MaintenanceWindow.TO_PB_FUNCTION));
+    }
+    if (region != null) {
+      zonePb.setRegion(region.toUrl());
+    }
+    return zonePb;
+  }
+
+  static Builder builder() {
+    return new Builder();
+  }
+
+  static Zone fromPb(com.google.api.services.compute.model.Zone zonePb) {
+    Builder builder = builder();
+    builder.zoneId(ZoneId.fromUrl(zonePb.getSelfLink()));
+    if (zonePb.getId() != null) {
```

Comment:
If/when you implement a "get" with options to specify fields, will ID a field that is always returned, regardless of whether the user specifies it or not?  If it will always be returned, I think `zonePb.getId()` should never be null, and we could remove this check.  (Also applies to other `fromPb`s where we check for null ID.)